### PR TITLE
Types for /util

### PR DESCRIPTION
**Pull request by colinmegill**
Adding typescript types to `src/util` folder

### CODE REVIEW

File: client/src/util/clip.ts [hunk 1]
```diff
@@ -7,11 +7,15 @@ If `setTo` is undefined, values less than `lower` will be set to `lower`,
...
+`setTo`. For instance, 
 
 */
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-export default function clip(arr: any, lower: any, upper: any, setTo: any) {
+export default function clip(
+  arr: Array<number>,
+  lower: number,
+  upper: number,
+  setTo: number | undefined
```
Comment (tihuan) [inline client/src/util/clip.ts:17]: shortcut for `number | undefined` is `setTo?: number` 🙆‍♂️

**Review by tihuan — COMMENTED** [review]
Typings look great! Thanks so much, @colinmegill 🎉 !

Can we also update the call sites that use these functions too? (Removing `eslint-disable` at the call sites 🙏 )

Thank you!

File: client/src/util/catLabelSort.ts [hunk 2]
```diff
@@ -11,8 +11,7 @@ TL;DR: sort order is:
 import isNumber from "is-number";
 import * as globals from "../globals";
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-function caseInsensitiveCompare(a: any, b: any) {
+function caseInsensitiveCompare(a: string, b: string) {
```
Comment (seve) [inline client/src/util/catLabelSort.ts:14]: We'll want to also be typing the return type

**Review by seve — CHANGES_REQUESTED** [review]

**Comment by tihuan** [comment]
@seve actually Brian just reminded me that Colin is out this week. So can you help take over this PR 😆 ? Thank you so much!!

**Comment by MillenniumFalconMechanic** [comment]
Hi @colinmegill, typings are looking good!

I am seeing and couple of compile errors when I run `npm run type-check` and lint errors when I run `npm run lint` locally:

#### npm run type-check
```
src/util/finiteExtent.ts:26:11 - error TS2532: Object is possibly 'undefined'.

26       if (min > val) min = val;
             ~~~

src/util/finiteExtent.ts:27:11 - error TS2532: Object is possibly 'undefined'.

27       if (max < val) max = val;
             ~~~

src/util/parseRGB.ts:27:12 - error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
  No index signature with a parameter of type 'string' was found on type '{}'.

27   let cv = colorCache[c];
              ~~~~~~~~~~~~~

src/util/parseRGB.ts:30:5 - error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
  No index signature with a parameter of type 'string' was found on type '{}'.

30     colorCache[c] = cv;
```

#### npm run lint

```
/Users/mim/git/cellxgene/client/src/util/clip.ts
  13:16  warning  Missing return type on function  @typescript-eslint/explicit-module-boundary-types

/Users/mim/git/cellxgene/client/src/util/finiteExtent.ts
  9:1  warning  Missing return type on function  @typescript-eslint/explicit-module-boundary-types

/Users/mim/git/cellxgene/client/src/util/parseBulkGeneString.ts
  9:16  warning  Missing return type on function  @typescript-eslint/explicit-module-boundary-types

/Users/mim/git/cellxgene/client/src/util/parseRGB.ts
  26:16  warning  Missing return type on function  @typescript-eslint/explicit-module-boundary-types

/Users/mim/git/cellxgene/client/src/util/scaleLinear.ts
  9:16  warning  Missing return type on function  @typescript-eslint/explicit-module-boundary-types

/Users/mim/git/cellxgene/client/src/util/scaleRGB.ts
  1:16  warning  Missing return type on function  @typescript-eslint/explicit-module-boundary-types
```

Are you able to see these errors and warnings with your setup?

Finally, I see there are a couple of remaining `any` error suppressions; I wanted to check if these are OK as is for this PR (and we can just update them iteratively going forward).

**Comment by tihuan** [comment]
OK looks like the linter is failing as expected now 💪

**Pull request closed without merging**